tasks/kmt.py

```diff
@@ -970,16 +984,20 @@ def build(
     assert len(domains) > 0, f"no vms found from list {vms}. Run `inv -e kmt.status` to see all VMs in current stack"
 
     cc.exec(
-        f"cd {CONTAINER_AGENT_PATH} && git config --global --add safe.directory {CONTAINER_AGENT_PATH} && inv -e system-probe.build --no-bundle",
+        f"cd {CONTAINER_AGENT_PATH} && git config --global --add safe.directory {CONTAINER_AGENT_PATH} && inv -e {component}.build --no-bundle",
     )
+
+    if component == "security-agent":  # security-agent build does not build ebpf files by default
+        cc.exec(f"cd {CONTAINER_AGENT_PATH} && inv -e system-probe.object-files")
```

Comment:
This has to run before the full build, since it creates certain files for runtime compilation used in the binary build.